aggregator/src/aggregator.rs

```diff
@@ -3360,16 +3361,74 @@ fn empty_batch_aggregations<
     .collect()
 }
 
-struct RequestBody<T> {
+#[derive(Clone)]
+struct RequestBody {
     content_type: &'static str,
-    request: T,
+    body: Bytes,
+}
+
+struct RequestTimer<'a> {
+    // Mutable state.
+    start: SyncMutex<Option<Instant>>,
+
+    // Immutable state.
+    http_request_duration_histogram: &'a Histogram<f64>,
+    domain: String,
+    endpoint: String,
+    method: String,
```

Comment:
nit: I suspect we could have `endpoint: &'static str` and avoid a copy. Harder for `method`, but since the value comes from [`Method::as_str`](https://docs.rs/reqwest/latest/reqwest/struct.Method.html#method.as_str), this could be `method: &'a str` without too much trouble.